Unexpected default X509 SVID TTL

PR #3445 added support for X509 and JWT specific SVID TTLs. As part of this, the `spire-server` binary config arguments were updated to include all of 
https://github.com/spiffe/spire/blob/d98577f7eb8862a660a91d3a8560bb4bbca7b083/cmd/spire-server/cli/run/run.go#L77-L79

The following lines set the config defaults for the X509 and JWT SVIDs
https://github.com/spiffe/spire/blob/d98577f7eb8862a660a91d3a8560bb4bbca7b083/cmd/spire-server/cli/run/run.go#L839-L840

The result is that the values for `DefaultX509SVIDTTL` and `DefaultJWTSVIDTTL` are always set to some value even if the startup config file does not set those. 

On face this seems reasonable, but an issue appears if the startup config instead sets `DefaultSVIDTTL` and neither of the new X509/JWT specific values.

In such a case the following blocks will always be executed and the desired `DefaultSVIDTTL` will be ignored.
https://github.com/spiffe/spire/blob/d98577f7eb8862a660a91d3a8560bb4bbca7b083/cmd/spire-server/cli/run/run.go#L472-L481
https://github.com/spiffe/spire/blob/d98577f7eb8862a660a91d3a8560bb4bbca7b083/cmd/spire-server/cli/run/run.go#L492-L501

The result is a breaking change if someone has set the `DefaultSVIDTTL` value and has now updated to version 1.5.0. 

For example, with this config file
```
server {
	...
	default_svid_ttl = "15m"
	...
}
```

the following will appear in the logs on startup
```
time="2022-11-07T19:46:50Z" level=warning msg="both default_x509_svid_ttl and default_svid_ttl are configured; default_x509_svid_ttl (1h0m0s) will be used for X509-SVIDs"
time="2022-11-07T19:46:50Z" level=warning msg="both default_jwt_svid_ttl and default_svid_ttl are configured; default_jwt_svid_ttl (5m0s) will be used for JWT-SVIDs"
```

These two lines in the test appear to be why this wasn't caught during testing. It was an issue I ran into during initial testing and resolved it incorrectly.

https://github.com/spiffe/spire/blob/d98577f7eb8862a660a91d3a8560bb4bbca7b083/cmd/spire-server/cli/run/run_test.go#L642-L643

Removing those two lines shows the test now failing as it should.
```
time="2022-11-07T15:08:10-05:00" level=warning msg="both default_x509_svid_ttl and default_svid_ttl are configured; default_x509_svid_ttl (1h0m0s) will be used for X509-SVIDs"
time="2022-11-07T15:08:10-05:00" level=warning msg="both default_jwt_svid_ttl and default_svid_ttl are configured; default_jwt_svid_ttl (5m0s) will be used for JWT-SVIDs"
.....

--- FAIL: TestNewServerConfig (0.01s)
    --- FAIL: TestNewServerConfig/default_svid_ttl_is_correctly_parsed (0.00s)
        run_test.go:646:
            	Error Trace:	/Users/dgove1/dev/spire/spire/cmd/spire-server/cli/run/run_test.go:646
            	            				/Users/dgove1/dev/spire/spire/cmd/spire-server/cli/run/run_test.go:1074
            	Error:      	Not equal:
            	            	expected: 1m0s
            	            	actual  : 1h0m0s
            	Test:       	TestNewServerConfig/default_svid_ttl_is_correctly_parsed
```

--- a/cmd/spire-server/cli/run/run.go
+++ b/cmd/spire-server/cli/run/run.go
@@ -469,7 +469,8 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
 		sc.AgentTTL = ttl
 	}
 
-	if c.Server.DefaultX509SVIDTTL != "" {
+	switch {
+	case c.Server.DefaultX509SVIDTTL != "":
 		ttl, err := time.ParseDuration(c.Server.DefaultX509SVIDTTL)
 		if err != nil {
 			return nil, fmt.Errorf("could not parse default X509 SVID ttl %q: %w", c.Server.DefaultX509SVIDTTL, err)
@@ -479,7 +480,7 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
 		if sc.X509SVIDTTL != 0 && c.Server.DefaultSVIDTTL != "" {
 			logger.Warnf("both default_x509_svid_ttl and default_svid_ttl are configured; default_x509_svid_ttl (%s) will be used for X509-SVIDs", c.Server.DefaultX509SVIDTTL)
 		}
-	} else if c.Server.DefaultSVIDTTL != "" {
+	case c.Server.DefaultSVIDTTL != "":
 		logger.Warn("field default_svid_ttl is deprecated; consider using default_x509_svid_ttl and default_jwt_svid_ttl instead")
 
 		ttl, err := time.ParseDuration(c.Server.DefaultSVIDTTL)
@@ -487,6 +488,10 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
 			return nil, fmt.Errorf("could not parse default SVID ttl %q: %w", c.Server.DefaultSVIDTTL, err)
 		}
 		sc.X509SVIDTTL = ttl
+	default:
+		// If neither new nor deprecated config value is set, then use hard-coded default TTL
+		// Note, due to back-compat issues we cannot set this default inside defaultConfig() function
+		sc.X509SVIDTTL = ca.DefaultX509SVIDTTL
 	}
 
 	if c.Server.DefaultJWTSVIDTTL != "" {
@@ -499,6 +504,10 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
 		if sc.JWTSVIDTTL != 0 && c.Server.DefaultSVIDTTL != "" {
 			logger.Warnf("both default_jwt_svid_ttl and default_svid_ttl are configured; default_jwt_svid_ttl (%s) will be used for JWT-SVIDs", c.Server.DefaultJWTSVIDTTL)
 		}
+	} else {
+		// If not set using new field then use hard-coded default TTL
+		// Note, due to back-compat issues we cannot set this default inside defaultConfig() function
+		sc.JWTSVIDTTL = ca.DefaultJWTSVIDTTL
 	}
 
 	if c.Server.CATTL != "" {
@@ -831,14 +840,12 @@ func checkForUnknownConfig(c *Config, l logrus.FieldLogger) (err error) {
 func defaultConfig() *Config {
 	return &Config{
 		Server: &serverConfig{
-			BindAddress:        "0.0.0.0",
-			BindPort:           8081,
-			CATTL:              ca.DefaultCATTL.String(),
-			LogLevel:           defaultLogLevel,
-			LogFormat:          log.DefaultFormat,
-			DefaultX509SVIDTTL: ca.DefaultX509SVIDTTL.String(),
-			DefaultJWTSVIDTTL:  ca.DefaultJWTSVIDTTL.String(),
-			Experimental:       experimentalConfig{},
+			BindAddress:  "0.0.0.0",
+			BindPort:     8081,
+			CATTL:        ca.DefaultCATTL.String(),
+			LogLevel:     defaultLogLevel,
+			LogFormat:    log.DefaultFormat,
+			Experimental: experimentalConfig{},
 		},
 	}
 }

--- a/cmd/spire-server/cli/run/run_test.go
+++ b/cmd/spire-server/cli/run/run_test.go
@@ -639,8 +639,6 @@ func TestNewServerConfig(t *testing.T) {
 			msg: "default_svid_ttl is correctly parsed",
 			input: func(c *Config) {
 				c.Server.DefaultSVIDTTL = "1m"
-				c.Server.DefaultX509SVIDTTL = ""
-				c.Server.DefaultJWTSVIDTTL = ""
 			},
 			test: func(t *testing.T, c *server.Config) {
 				require.Equal(t, time.Minute, c.X509SVIDTTL)

